### PR TITLE
wallet-api: Tutorial

### PR DESCRIPTION
**Pull request by j-mueller**
* links to haddocks
* (WalletAPI m) instead of MockWallet
* Explain intervals
* Comments and clarifications

### CODE REVIEW

File: wallet-api/tutorial/Tutorial.md [hunk 1]
```diff
@@ -157,7 +156,10 @@ Then we compute the total value of all transaction inputs, using `P.foldr` on th
 ```haskell
                   totalInputs :: Int
                   totalInputs =
+                      -- define a function "v" that extracts the ada value from a 'PendingTxIn'
                       let v (PendingTxIn _ _ (Value vl)) = vl in
+
+                      -- Apply "v" to each transaction input, summing up the results
                       $$(P.foldr) (\i total -> total + v i) 0 ins
```
Comment (michaelpj) [inline wallet-api/tutorial/Tutorial.md]: You could just do the pattern match in the lambda argument and then you wouldn't need the helper function.

File: wallet-api/tutorial/Tutorial.md [hunk 2]
```diff
@@ -305,12 +307,14 @@ collectFundsTrigger c = W.andT
     (W.slotRangeT (W.interval (endDate c) (collectionDeadline c)))
 ```
 
+`fundsAtAddressT` and `slotRangeT` take `Interval Value` and `Interval Slot` arguments respectively. The [`Interval`](https://input-output-hk.github.io/plutus/wallet-api-0.1.0.0/html/Wallet-API.html#t:Interval) type is part of the `wallet-api` package. The [`Ledger.Interval`](https://input-output-hk.github.io/plutus/wallet-api-0.1.0.0/html/Ledger-Interval.html#v:Interval) module that originally defines it illustrates how to write a data type and associated operations that can be used both in off-chain and in on-chain code.
```
Comment (michaelpj) [inline wallet-api/tutorial/Tutorial.md]: Cute, I hadn't noticed you were also using `Interval Value` for fund ranges.

File: wallet-api/tutorial/Tutorial.md [hunk 3]
```diff
@@ -136,15 +135,15 @@ Before we check whether `act` is permitted, we define a number of intermediate v
                   signedBy (PubKey pk) (Signature s) = pk == s
 ```
 
-There is no standard library of functions that are automatically in scope for on-chain code, so we need to import the ones that we want to use from the `Validation` module using the `\$\$()` splicing operator.
+There is no standard library of functions that are automatically in scope for on-chain code, so we need to import the ones that we want to use from the [`Ledger.Validation`](https://input-output-hk.github.io/plutus/wallet-api-0.1.0.0/html/Ledger-Validation.html) module using the `$$()` splicing operator. [`Ledger.Validation`](https://input-output-hk.github.io/plutus/wallet-api-0.1.0.0/html/Ledger-Validation.html) contains a subset of the standard Haskell prelude, exported as Template Haskell quotes. Code can only be imported from other libraries into validator scripts if it is available as a Template Haskell splice (so we can use `$$()` to splice it in).
```
[the remark this answers is not in the document]
Comment (j-mueller) [reply wallet-api/tutorial/Tutorial.md]: you're right. I changed it.

File: wallet-api/tutorial/Tutorial.md (hunk 1 above)
Comment (j-mueller) [reply wallet-api/tutorial/Tutorial.md]: True, but I think this is slightly more readable.

**Pull request merged**